app/services/promo_registrar.rb

```diff
@@ -12,44 +15,47 @@ def perform
     channels.each do |channel|
       if should_register_channel?(channel)
         referral_code = register_channel(channel)
-        promo_registration = PromoRegistration.new(channel_id: channel.id, promo_id: active_promo_id, referral_code: referral_code)
+        promo_registration = PromoRegistration.new(channel_id: channel.id, promo_id: @promo_id, referral_code: referral_code)
         promo_registration.save!
```

Comment:
when referral_code is nil i believe this will raise and stop code generation for subsequent channels